backend/battle/forms.py

```diff
@@ -1,24 +1,76 @@
-# from django import forms
+from django import forms
+from pokemon.models import Pokemon
+from battle.models import Battle, PokemonTeam, Team
+from battle.battles.battle import validate_sum_pokemons
 
-# from .models import Battle
 
+class BattleForm(forms.ModelForm):
+    class Meta:
+        model = Battle
+        fields = ['opponent', ]
 
-# class BattleForm(forms.ModelForm):
+    def clean(self):
+        cleaned_data = super().clean()
+        if cleaned_data['opponent'] == self.initial['creator']:
```

Comment:
It's because I don't have the creator value on cleaned_data by default. I put the creator value on form_valid(). So to solve this I pass the self.request.user by get_initial()

Reply:
I fixed it :) thanks